diff --git a/common.sh b/common.sh
index 9eee947..5d707a3 100755
--- a/common.sh
+++ b/common.sh
@@ -40,8 +40,10 @@ build_kernel()
 
 		if [ "${V}" = "guest" ]; then
 			BRANCH="${KERNEL_GUEST_BRANCH}"
+      GIT_COMMIT="${KERNEL_GUEST_COMMIT}"
 		else
 			BRANCH="${KERNEL_HOST_BRANCH}"
+      GIT_COMMIT="${KERNEL_HOST_COMMIT}"
 		fi
 
 		# If ${KERNEL_GIT_URL} is ever changed, 'current' remote will be out
@@ -68,7 +70,8 @@ build_kernel()
 
 		pushd ${V} >/dev/null
 			run_cmd git fetch current
-			run_cmd git checkout current/${BRANCH}
+			# run_cmd git checkout current/${BRANCH}
+      run_cmd git checkout ${GIT_COMMIT}
 			COMMIT=$(git log --format="%h" -1 HEAD)
 
 			run_cmd "cp /boot/config-$(uname -r) .config"
@@ -175,7 +178,9 @@ build_install_ovmf()
 
 	pushd ovmf >/dev/null
 		run_cmd git fetch current
-		run_cmd git checkout current/${OVMF_BRANCH}
+		# run_cmd git checkout current/${OVMF_BRANCH}
+    run_cmd git checkout ${OVMF_COMMIT}
+    run_cmd git apply ${OVMF_PATCH}
 		run_cmd git submodule update --init --recursive
 		run_cmd make -C BaseTools
 		. ./edksetup.sh --reconfig
@@ -214,7 +219,8 @@ build_install_qemu()
 
 	pushd qemu >/dev/null
 		run_cmd git fetch current
-		run_cmd git checkout current/${QEMU_BRANCH}
+		# run_cmd git checkout current/${QEMU_BRANCH}
+    run_cmd git checkout ${QEMU_COMMIT}
 		run_cmd ./configure --target-list=x86_64-softmmu --prefix=$DEST
 		run_cmd $MAKE
 		run_cmd $MAKE install
diff --git a/stable-commits b/stable-commits
index bc851d1..b428ec6 100644
--- a/stable-commits
+++ b/stable-commits
@@ -3,13 +3,16 @@
 #
 
 # hypervisor commit
-KERNEL_GIT_URL="https://github.com/AMDESE/linux.git"
+KERNEL_GIT_URL="https://github.com/dimstav23/amd-linux.git"
 KERNEL_HOST_BRANCH="snp-host-latest"
+KERNEL_HOST_COMMIT="6b293770dac2fc37e7a880a321045d54bc88b0ce"
 KERNEL_GUEST_BRANCH="snp-guest-latest"
+KERNEL_GUEST_COMMIT="98543c2aa64942a32cf623f2b8b3ddd01ef515c2"
 
 # qemu commit
-QEMU_GIT_URL="https://github.com/AMDESE/qemu.git"
+QEMU_GIT_URL="https://github.com/dimstav23/amd-qemu.git"
 QEMU_BRANCH="snp-latest"
+QEMU_COMMIT="b6ee1218e6c9b98a556841615dd10d094e648393"
 
 # guest bios
 #   An AP creation fix added after the 'edk2-stable202302' tag/release is
@@ -19,5 +22,7 @@ QEMU_BRANCH="snp-latest"
 #   please use the 'sev-snp-legacy' branch of this AMDSEV repo instead,
 #   which will build the latest known-working QEMU/OVMF trees for older
 #   SNP hypervisor/host kernels.
-OVMF_GIT_URL="https://github.com/AMDESE/ovmf.git"
+OVMF_GIT_URL="https://github.com/dimstav23/amd-ovmf.git"
 OVMF_BRANCH="snp-latest"
+OVMF_COMMIT="09fbe92dc545779671d7fd89a5bd4f1b14f7e69b"
+OVMF_PATCH="../../ovmf.patch"
\ No newline at end of file
