diff --git a/Makefile b/Makefile
index 3e46cd2..4caa901 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 
 GCC		= gcc
 
-SHELL		:= /bin/bash
+SHELL		:= ${SHELL}
 
 A_FLAGS		:= -D__ASSEMBLY__
 
@@ -57,7 +57,7 @@ prereq: .prereq
 
 .prereq:
 	curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
-	source $(HOME)/.cargo/env
+	#source $(HOME)/.cargo/env
 	echo "source $(HOME)/.cargo/env" >> ~/.bashrc
 	rustup component add rust-src
 	rustup component add llvm-tools-preview
diff --git a/scripts/common.sh b/scripts/common.sh
index 3d39ce7..2783d7a 100755
--- a/scripts/common.sh
+++ b/scripts/common.sh
@@ -61,7 +61,8 @@ build_kernel()
 				run_cmd git checkout current/${BRANCH}
 				COMMIT=$(git log --format="%h" -1 HEAD)
 
-				run_cmd "cp /boot/config-$(uname -r) .config"
+				#run_cmd "cp /boot/config-$(uname -r) .config"
+				run_cmd "zcat /proc/config.gz > .config"
 				run_cmd ./scripts/config --set-str LOCALVERSION "$VER-$COMMIT"
 				run_cmd ./scripts/config --disable LOCALVERSION_AUTO
 				run_cmd ./scripts/config --enable  DEBUG_INFO
diff --git a/scripts/launch-qemu.sh b/scripts/launch-qemu.sh
index 58e55df..ac30df3 100755
--- a/scripts/launch-qemu.sh
+++ b/scripts/launch-qemu.sh
@@ -19,6 +19,8 @@ SVSM=""
 CONSOLE="serial"
 UEFI_BIOS_CODE="./usr/local/share/qemu/OVMF_CODE.fd"
 UEFI_BIOS_VARS="./usr/local/share/qemu/OVMF_VARS.fd"
+#UEFI_BIOS_CODE="/run/libvirt/nix-ovmf/OVMF_CODE.fd"
+#UEFI_BIOS_VARS="/run/libvirt/nix-ovmf/OVMF_VARS.fd"
 BIOS_DEBUG=""
 VNC_PORT=""
 ALLOW_DEBUG=""
@@ -29,6 +31,8 @@ SNP_FLAGS="0"
 
 QEMU_INSTALL_DIR="./usr/local/bin/"
 
+echo $LD_LIBRARY_PATH
+
 usage() {
 	echo "$0 [options]"
 	echo "Available <commands>:"
